Update README.md to use latest HA terminology & automation notation

gcoan opened the pull request:
Aligned the readme to align to HA 2024.10:

- Changed the sample automations to pluralise trigger, condition and action sections
- Changed platform to trigger
- Changed all references from calling a ‘service’ to calling an ‘action’

BJReplay reviewed (changes requested):
I haven't checked the example automations to see if they are correct or not: assuming that they are, thank you!

However, it looks like you have done a **change all** "service" to "action", and this has changed previous change notes.

That isn't appropriate: previous changes did use the term service call, not action call, and that doesn't make sense.

Please review, and only change those cases where it is essential.

gcoan commented:
> However, it looks like you have done a change all "service" to "action", and this has changed previous change notes.

I actually went through each line and manually changed the terminology where it made sense to do so.

On the change history, yes I deliberately changed all the references to 'service' to 'action' for consistency and alignment with the current HA terminology, even though that is rewriting history I thought that would be clearer than having references to service calls when that's not a term being used any more. 
Happy to revert those history changes back if you think we should keep the history as is

Cheers

BJReplay commented:
Yeah, definitely needs fixing - it is an action, not an action call.

Service call was what it was (and all the old references should still say that).

They still **are services** - https://github.com/BJReplay/ha-solcast-solar/blob/main/custom_components/solcast_solar/services.yaml defines the _actions_ that you call from automations 🤣

BJReplay commented:
> They still are services

Which is to say that the non-example text will still be subject to contemplation.

Just because the HA people have decided to call them actions doesn't mean that we've got to do a miniplenty on this...

gcoan commented:
OK I'll have a trawl through and revise them.

autoSteve reviewed (changes requested):
Agree with re-writing history, not, but love the current state modifications. Thank you Geoff. The code needs updating to reflect this. #todo.

gcoan commented:
@BJReplay and @autoSteve have been through the changes again, reverted the change history revisions and fixed references to 'action calls' to correct terminology (generally it is now just 'action').
Using HA2024.8 announcement https://developers.home-assistant.io/blog/2024/07/16/service-actions/ to guide the rewording.

And I did test the automations and they are all fine.  I was confident they were, but worth double checking.

BJReplay commented:
Thanks @gcoan, really starting to come together.

Release notes mentioning this change look wrong though as 4.2.3 is currently out, but that might just be me reading the compare on my phone with dodgy eyes.

gcoan commented:
> Thanks @gcoan, really starting to come together.
> 
> Release notes mentioning this change look wrong though as 4.2.3 is currently out, but that might just be me reading the compare on my phone with dodgy eyes. 

There wasn't any release notes for anything after 4.2.0, so I added the entry for 4.2.3 assuming you'd just merge this in to the current release (as its documentation only) rather than hold it off for 4.2.4 or whatever.  But no issue to change or even remove entirely the release notes line

BJReplay commented:
Good point - but there is a release coming up, and even though it is only readme, it contains code. @autoSteve this feels like v4.2.4 for multiple hard limits and readme changes?

autoSteve commented:
FYI, I forgot to update the release history.

I have updated it as part of PR #198. @gcoan if you change this area you will cause a conflict.

<img width="1699" alt="image" src="https://github.com/user-attachments/assets/b8e1b30c-428d-4c23-bf69-7db7c80aee29">

autoSteve commented:
> this feels like v4.2.4 for multiple hard limits and readme changes

That's where I'm heading as a suggestion @BJReplay.

Pending anything else exploding unexpectedly...

BJReplay commented:
We'll work out a way to get both in.  Even if we put in the action changes post release shortly after the 4.2.4 release as 4.2.5.

@gcoan time to learn your git rebase skilz :)

BJReplay commented on README.md:
```diff
@@ -706,7 +708,7 @@ Full Changelog: https://github.com/BJReplay/ha-solcast-solar/compare/v4.2.0...v4
 v4.2.0
 * Generally available release of v4.1.8 and v4.1.9 pre-release features
-* Translations of service call error responses by @autoSteve
+* Translations of services call error responses by @autoSteve
```
should be service (not services)

gcoan replied:
changed

BJReplay commented on README.md:
```diff
@@ -696,6 +695,9 @@ series:
 
 ## Changes
 
+v4.2.3
```
Suggest removing this section.  That we we can merge this into a new branch, and merge Steve's changes into a new branch, and both can go into a single release.  In any case, remove it will fix the clash with the existing v4.2.3.

gcoan replied:
Yeah, probably simpler that I do that

BJReplay commented on README.md:
```diff
@@ -447,7 +446,7 @@ The scenario requiring use of this limit is straightforward, but note that hardl
 Consider a scenario where you have a single 6kW string inverter, and attached are two strings each of 5.5kW potential generation pointing in separate directions. This is considered "over-sized" from an inverter point of view. It is not possible to set an AC generation limit for Solcast that suits this scenario when configured as two sites, as in the mid-morning or afternoon in Summer a string may ...
 
-The hard limit may be set in the integration configuration, or set via a service call in `Developer Tools`.
+The hard limit may be set in the integration configuration, or set via manually invoking the Action in `Developer Tools`.
```
Prefer lower case action.  Yes, it might be capitalised in the, but unless highlighted (e.g. with backticks - not recommended), just refer to it as an action.

gcoan replied:
done

BJReplay commented on README.md:
```diff
@@ -390,7 +389,7 @@ Example of half-hourly dampening for all sites:
 When calculating dampening using an automation it may be beneficial to use undampened forecast values as input.
 
-This is possible by using the service call `solcast_solar.query_forecast_data`, and including `undampened: true` in the call. If using granular dampening then the site may also be included in the call.
+This is possible by using the action `solcast_solar.query_forecast_data`, and including `undampened: true` in the parameters. If using granular dampening then the site may also be included in the action.
```
second action should probably also be parameters - but will defer to @autoSteve for both.

gcoan replied:
added second 'parameters'

BJReplay commented on README.md:
```diff
@@ -310,7 +309,7 @@ Dampening is applied to future forecasts whenever a forecast is fetched, so fore
 > Retained dampened historical forecasts is a recent change, and may require automation modification to read undampened forecast history instead. See [Reading forecast values in an automation](#reading-forecast-values-in-an-automation) and [Changes](#changes) below.
 
-Per-site and per-half hour dampening is possible only by using service calls or modifying a dampening configration file. See [Granular dampening](#granular-dampening) below.
+Per-site and per-half hour dampening is possible only by using the integration action or modifying a dampening configration file. See [Granular dampening](#granular-dampening) below.
```
drop integration

gcoan replied:
I was trying to give more context as to the 'action'.  I've put the correct action name in now which makes it clearer

BJReplay reviewed (changes requested):
I've had a chance to review in detail and made a few more requested changes.

gcoan commented:
> @gcoan time to learn your git rebase skilz :)

I'd rather not.

Try number 3 ...
